### PR TITLE
Jungle Plants

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
From Ausops

ausops

### CODE REVIEW

**Comment by chanoc1** [comment]
looks nice 👍

**Comment by Repukan** [comment]
Sexy

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -306,3 +313,61 @@
...
+	icon_state = "busha"
+
+/obj/structure/flora/junglebush/New()
+	icon_state = "[icon_state][rand(1, 3)]"
+	..()
+
+/obj/structure/flora/junglebush/b
+	icon_state = "bushb"
+
+/obj/structure/flora/junglebush/b/New()
```
Comment (RemieRichards) [inline code/game/objects/structures/flora.dm]: these extra ```New()```s aren't needed.

**Comment by monster860** [comment]
I think Im gonna steal those sprites for FTL13

**Comment by FantasticFwoosh** [comment]
One criticism, if those plants are just turf colouring. Aren't the larger models going to look bad when they clip against buildings or solid constructions/

**Comment by Ausops** [comment]
to an extent, clever mapping can mitigate it effectively enough

**Comment by Steelpoint** [comment]
Won't be a issue as @Ausops noted. Since we're going with pre-made maps the mappers themselves should easily be able to avoid this issue.

**Comment by Jordie0608** [comment]
Since these layer above mobs etc. and I assume are entirely cosmetic it might be good to have their `mouse_opacity` be 0 to avoid any problems with atoms behind them; not that they occlude much.
(Unless layering is reverse to how I remember)

**Comment by Militaires** [comment]
@korphaeron include this broad-leafed one http://i.imgur.com/9TaX0H3.png

**Comment by KorPhaeron** [comment]
@Jordie0608 

That would kill tactical plants

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -358,4 +358,16 @@
...
 /obj/structure/flora/junglebush/c
-	icon_state = "bushc"
\ No newline at end of file
+	icon_state = "bushc"
+
+/obj/structure/flora/junglebush/large
+	icon_state = "bush"
+	icon = 'icons/obj/flora/largejungleflora.dmi'
+	pixel_x = -16
+	layer = FLY_LAYER
```
Comment (ChangelingRain) [inline code/game/objects/structures/flora.dm]: ABOVE_ALL_MOB_LAYER
Or WALL_OBJ_LAYER, if you want large mobs to appear above these.

File: code/game/objects/structures/flora.dm [hunk 3]
```diff
@@ -306,3 +313,61 @@
...
+
+
+/obj/structure/flora/rock/pile/largejungle
+	name = "rocks"
+	icon = 'icons/obj/flora/largejungleflora.dmi'
+	density = 0
```
Comment (lzimann) [inline code/game/objects/structures/flora.dm]: FALSE

File: code/game/objects/structures/flora.dm [hunk 4]
```diff
@@ -79,6 +79,13 @@
 	icon_state = "tree_[rand(1, 6)]"
 	..()
 
+/obj/structure/flora/tree/jungle
+	name = "tree"
+	desc = "It's seriously hampering your view of the jungle."
+
+/obj/structure/flora/tree/jungle/New()
```
Comment (lzimann) [inline code/game/objects/structures/flora.dm]: Use Initialize() instead of New()

**Comment by Jordie0608** [comment]
@KorPhaeron

**Comment by Steelpoint** [comment]
I tested in game, the Jungle Trees don't have the single code line that actually tells them to get their mob file from the 'jungletrees.dmi' file, leading to the game spawning a table with a smiling face in game.

**Comment by Steelpoint** [comment]
Also I can't actually see the tree in game even when I fix the issue.

**Comment by Steelpoint** [comment]
Also the rocks suffer the same problem.

**Comment by PKPenguin321** [comment]
Planetstation off to a good start

**Pull request merged**